apicapi/apic_client.py

```diff
@@ -515,10 +566,51 @@ def login(self, usr=None, pwd=None):
                                          err_text=attributes['text'],
                                          err_code=attributes['code'])
 
+    def set_private_key(self, usr, cert_name, private_key_file):
+        """Set the X.509 private key used for session-less REST calls"""
+        url = self._api_url('mo/%s' % self._get_cert_dn(usr, cert_name))
+
+        try:
+            with open(private_key_file) as key_file:
+                private_key = load_privatekey(FILETYPE_PEM, key_file.read())
+        except Exception, e:
+            LOG.error("Failed to load private key from file %s: %s" %
+                      (private_key_file, e))
+            raise
+
+        old_username = self.username
+        old_cert_name = self.cert_name
+        old_private_key = self.private_key
+
+        self.username = usr
+        self.cert_name = cert_name
+        self.private_key = private_key
+        # Verify that the cert-name and key are ok
+        try:
+            response = self._do_request(ApicSession.HTTP_GET, url)
+        except rexc.Timeout:
```

Comment:
I've removed restoring of old values. The user of the class can call set_private_key again if they see failures.